src/ui_manager.cpp

```diff
@@ -421,14 +454,25 @@ void ui_adaptor::redraw_invalidated()
                         ui.invalidated = false;
                     }
                 }
+                if( ui.is_imgui == draw_imgui && ( !ui.is_imgui || imgui_is_on_top ) ) {
+                    ui.redraw();
+                }
             }
             if( !restart_redrawing && cursor_pos.has_value() ) {
                 restore_cursor( cursor_pos.value() );
             }
         }
     } while( restart_redrawing );
+
+#if !(defined(TILES) || defined(_WIN32))
+    ImGui::Render();
+
+    ImTui_ImplText_RenderDrawData( ImGui::GetDrawData(), imtui_screen );
+    ImTui_ImplNcurses_DrawScreen();
+#endif
```

Comment:
Does this refresh the game window? If so this should be placed in `refresh_display` instead.

Reply:
I do it this way because the drawing strategy is slightly different between SDL and Curses builds. On Curses builds, when you call handle_input, refresh_display is not called. If I call refresh_display every loop of a UI window, it will result in a double draw on SDL which is not acceptable.

Reply:
> On Curses builds, when you call input_context::handle_input, refresh_display is not called.

It is. Or rather, `catacurses::doupdate` is called which is synonymous to `refresh_display` in both the SDL and curses builds.

> If I call refresh_display every loop of a UI window, it will result in a double draw on SDL which is not acceptable.

I mean, the function of `ui_manager::redraw` is to update the framebuffer but not display it on-screen, so you should place these code in `refresh_display` instead. Since `refresh_display` is synonymous to `catacurses::doupdate` I think it will work without adding new calls to `refresh_display` and therefore no double draw will happen.

Reply:
>It is. Or rather, catacurses::doupdate is called which is synonymous to refresh_display in both the SDL and curses builds.

OK, I see. I will move all the imgui redraw calls into doupdate then

Reply:
I actually thought about this a bit more. The ImTui calls do not update the screen, they write to stdscr (if there is anything to render) but they do not refresh the terminal screen. Maybe we could slow down the number of calls to doupdate to limit the CPU usage of ImTui.